doc/semantics.md

```diff
@@ -48,44 +60,80 @@ type they were created with. The following are examples:
 - 1 matches `Byte`, `Short`, `Int`, `Float`, `Double`
 - 128 (`> Byte.MaxValue`) matches `Short`, `Int`, `Float`, `Double`
 - 32768 (`> Short.MaxValue`) matches `Int`, `Float`, `Double`
+- 2147483647 matches `Int`, `Double` if strict-floats are enabled,
+  otherwise `Float` as well
 - 2147483648 (`> Int.MaxValue`) matches `Float`, `Double`
-- 1.2 matches `Float`, `Double`
+- 1.5 matches `Float`, `Double`
+- 1.4 matches `Double` only if strict-floats are enabled,
+  otherwise `Float` and `Double`
+- `NaN`, `Infinity`, `-Infinity` and `-0.0` match `Float`, `Double`
+
+As a consequence, the following apparent subtyping relationships hold:
+
+    Byte <:< Short <:<  Int  <:< Double
+                   <:< Float <:<
 
-As a consequence, the following apparent subtyping relationship holds:
+if strict-floats are enabled, or
 
     Byte <:< Short <:< Int <:< Float =:= Double
 
-### toString for integral Floats and Doubles
-Calling `toString` on a Float or a Double that holds an integral
-value, will not append ".0" to that value:
+otherwise.
 
-{% highlight scala %}
-println(1.0)
-// Scala:    1.0
-// Scala.js: 1
-{% endhighlight %}
+## Undefined behaviors
 
-This is due to how numeric values are represented at runtime in
-Scala.js. Use a formatting interpolator if you always want to show
-decimals:
+The JVM is a very well specified environment, which even specifies how some
+bugs are reported as exceptions.
+Some examples are:
 
-{% highlight scala %}
-val x = 1.0
-println(f"$x%.1f")
-// Scala:    1.0
-// Scala.js: 1.0
-{% endhighlight %}
+* `NullPointerException`
+* `ArrayIndexOutOfBoundsException` and `StringIndexOutOfBoundsException`
+* `ClassCastException`
+* `ArithmeticException` (such as integer division by 0)
+* `StackOverflowError` and other `VirtualMachineError`s
+
+Because Scala.js does not receive VM support to detect such erroneous
+conditions, checking them is typically too expensive.
+
+Therefore, all of these are considered
+[undefined behavior](http://en.wikipedia.org/wiki/Undefined_behavior).
+
+Some of these, however, can be configured to be compliant with sbt settings.
```

Comment:
to be compliant with the JVM specification using sbt settings. (or just using sbt settings).